Fix to make the click event use the correct target for thumbnails

MusikAnimal opened the pull request:
Thumbnail images are tokenized but are usually floated to the left or
right, meaning the element that triggered the event may differ from the
visible element that triggered the event (the one with the token ID).
Additionally, we need to attach the popup to the thumbnail itself.

Bug: T231959

samwilson reviewed (approved):
Sorry, I thought I'd reviewed this last week. Looks good to me. Just needs to be rebased I think.

MusikAnimal committed:
Fix to make the click event use the correct target for thumbnails

Thumbnail images are tokenized but are usually floated to the left or
right, meaning the element that triggered the event may differ from the
visible element that triggered the event (the one with the token ID).
Additionally, we need to attach the popup to the thumbnail itself.

Bug: T231959

MusikAnimal commented:
Rebased! Merging based on Sam's approval.
